source/server/filter_chain_manager_impl.h

```diff
@@ -107,10 +111,19 @@ class FilterChainImpl : public Network::DrainableFilterChain {
     factory_context_ = std::move(filter_chain_factory_context);
   }
 
+  bool isFakeFilterChain() const { return is_fake_placeholder_; }
+
+  const envoy::config::listener::v3::FilterChain* const& getFilterChainMessage() const {
+    return filter_chain_message_;
+  }
+
 private:
   Configuration::FilterChainFactoryContextPtr factory_context_;
   const Network::TransportSocketFactoryPtr transport_socket_factory_;
   const std::vector<Network::FilterFactoryCb> filters_factory_;
+
+  const envoy::config::listener::v3::FilterChain* const filter_chain_message_;
+  bool is_fake_placeholder_;
```

Comment:
Inside a fake filter chain, we store the protobuf message config and mark it as ```is_fake_placeholder_```.